java/src/main/native/src/AggregationJni.cpp

```diff
@@ -34,58 +34,47 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Aggregation_close(JNIEnv *env,
 JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Aggregation_createNoParamAgg(JNIEnv *env,
                                                                          jclass class_object,
                                                                          jint kind) {
+
   try {
     cudf::jni::auto_set_device(env);
     std::unique_ptr<cudf::aggregation> ret;
-    // These numbers come from Aggregation.java and must stay in sync
-    switch (kind) {
-      case 0: // SUM
+    switch (static_cast<cudf::aggregation::Kind>(kind)) {
```

Comment:
The problem with the change you've made here is now this loop is at the whim of the ordering in a cudf header file rather than being controlled within the JNI bindings source.  Using a separate enumeration was not an accident and was done to solve a problem.  If someone in libcudf adds a new value in the middle of the enumeration, which has happened multiple times before, then this new code suddenly, subtly, breaks and misaligns the aggregations from what Java thinks it is doing.  IMO this change should not be made, and it should revert to switching on the numbers that came from the Java file.

This comment applies to all the places where the `kind` integer being passed from the JVM is being statically cast to the libcudf kind.

Reply:
Oh no. So the fixed numbers here are corresponding to the enum in `java/src/main/java/ai/rapids/cudf/Aggregation.java`, not in cudf. OMG....
Should we have any way to import C++ enum into java?

Reply:
> Using enum directly from C++ header, instead of having a (manual) mirrored enum, should be favored and that can eliminate all the related problems.

Yes, using the enum directly would be great, _but if and only if the Java code is guaranteed to be in sync with the C++ enum_.  The way it's coded in this PR is purely worse, because routinely libcudf code is changed without updating the corresponding Java code.

Note that in this case, it's not required to pass the exact same C++ ID from Java to the JNI layer. It does not have to match the libcudf enum because we're not calling libcudf with that enum.  Would it be nice if they matched?  Yes, but it's not critical that they do.

If we can get the Java enums to be programmatically generated from the C++ enum that would be great, but IMO should not be part of this PR.

Reply:
@jlowe Thanks for reviewing this PR. I have reverted the change of enumeration, and fixed other problems you pointed out.